### ISSUE TITLE
Deterministic resampling

### ISSUE THREAD

**Issue by hinto-janai** [HEnquist/rubato#73]
Given a `rubato::FftFixedIn` with the same input for:
- `new()`
- `process_into_buffer()`

with the same buffer capacities, is the resampling process deterministic?

i.e. will 2 instances of resampling with the same inputs have the same output every time? (excluding small float differences)

**Comment by HEnquist** [comment]
Yes, if you create a FftFixedIn and call process a couple of times to process some data, the repeating the whole thing again will produce the same result again.

Note that if you feed the same data to a FftFixedIn resampler twice, without recreating or resetting the resampler in between, then you won't get the exact same result because this resampler has an internal buffer. This is needed because the resampler in most cases works with a different chunk size internally than the input size. FftFixedInOut on the other hand does not have this buffer, but for that one you cannot choose the exact input or output chunk sizes freely.

**Comment by hinto-janai** [comment]
Thanks, could I add tests for this?

**Comment by HEnquist** [comment]
What kind of test did you have in mind? I'm not sure there is much value in testing this. Being deterministic is basically by definition and will only change if the resampler has some strange bug giving random output. This is already being tested.
